### DISCUSSION TITLE
Radarr not displaying info - possibly due to large "Missing" count?

### DISCUSSION THREAD

**Discussion by jwfox5150** [linuxserver/Heimdall#739] [App issues]
Heimdall 2.2.2
Radarr 4.0.4.5922

Testing the API connection indicates that it is working.
A 2nd instance of Radarr with a small (6) missing count displays correctly.

My primary instance has close to 2,000 missing, and I believe this may be the cause of Heimdall not displaying stats.

**Comment by jwfox5150** [comment]
Following up: leaving Heimdall on the tag page where Radarr exists *eventually* loads stats from this instance. Recommend caching API stats to prevent extremely long load time, or perhaps changing the API call to Radarr to generate stats quicker.

**Comment by JMCee** [comment]
I have the same issue with Radarr missing stats. The API connection fails for me even though I've not changed any settings. Radarr had an update on the 31st January and the update notes state "Remove Legacy API". Not sure if that could be the cause?

Adding that I only have two releases missing and I don't see stats.

**Comment by jwfox5150** [comment]
> The API connection fails for me

Our issues sound like two separate issues. My API connection is working

**Comment by CJ7SHADOW** [comment]
> I have the same issue with Radarr missing stats. The API connection fails for me even though I've not changed any settings. Radarr had an update on the 31st January and the update notes state "Remove Legacy API". Not sure if that could be the cause?
> 
> Adding that I only have two releases missing and I don't see stats.

You can fix the API connection by editing the Radarr.php file located (for me on Truenas anyways) at usr/local/www/heimdall/app/SupportedApps/Radarr

the last line of code should be : $api_url = parent::normaliseurl($this->config->url).'api/'.$endpoint.'?apikey='.$this->config->apikey;
        return $api_url;

You need to add "v3/" without the quotes after api/
 so it should look like this: $api_url = parent::normaliseurl($this->config->url).'api/v3/'.$endpoint.'?apikey='.$this->config->apikey;
        return $api_url;

**Comment by JMCee** [comment]
> > I have the same issue with Radarr missing stats. The API connection fails for me even though I've not changed any settings. Radarr had an update on the 31st January and the update notes state "Remove Legacy API". Not sure if that could be the cause?
> > Adding that I only have two releases missing and I don't see stats.
> 
> You can fix the API connection by editing the Radarr.php file located (for me on Truenas anyways) at usr/local/www/heimdall/app/SupportedApps/Radarr
> 
> the last line of code should be : $api_url = parent::normaliseurl($this->config->url).'api/'.$endpoint.'?apikey='.$this->config->apikey; return $api_url;
> 
> You need to add "v3/" without the quotes after api/ so it should look like this: $api_url = parent::normaliseurl($this->config->url).'api/v3/'.$endpoint.'?apikey='.$this->config->apikey; return $api_url;

Yep, that fixed the issue and brought the stats back. Thank you! The file location on Unraid is /mnt/cache/appdata/heimdall/www/SupportedApps/Radarr

**Comment by rackey12** [comment]
i am having the same issue. i have found the Radarr.php file and updated the URL but i can still not get the stats to show. i am using docker confirmed that editing the file has been saved and also recreated the heimdall container and the radarr container.
how can i check the API connection in heimdall?

**Comment by DaVoodoo** [comment]
Same issue here. Before I added the "/v3" to the file, the API key would fail. After I added the "/v3" the API key is now valid but the stats still don't show up. Anything else I could try?

EDIT: Just rebooted my nas and it is working now.

**Comment by Brandoskey** [comment]
I've been having this issue for some months now. /v3 was already in the php file, no amount of restarts fix the issue
